components/Availability/AvailabilityCalendar.jsx

```diff
@@ -0,0 +1,170 @@
+import { View, Text, ScrollView, Pressable } from "react-native";
+import { useState, useRef, useEffect } from "react";
+import DateTimePicker from "@react-native-community/datetimepicker";
+import DateCircles from "./DateCircles";
+
+const splitDate = (date) => {
+    year = date.getFullYear();
+    month = date.getMonth() + 1;
+    day = date.getDate();
+    return [year, month, day];
+};
+
+const editDate = (oldDate, daysToChange) => {
+    newDate = new Date(oldDate);
+    newDate.setTime(newDate.getTime() + daysToChange * 24 * 60 * 60 * 1000);
+    return newDate;
+};
+
+const findColour = (colour) => {
+    if (colour != "none") {
+        return "bg-" + colour + "-400";
+    } else return "bg-white";
+};
+
+export default function AvailabilityCalendar({ availabilityInfo, setAvailabilityInfo }) {
+    const [date, setDate] = useState(new Date());
+    const scrollViewRef = useRef();
+    const handleOnPress = (hour, daysToChange) => {
+        const correctDate = editDate(date, daysToChange);
+        const dayData = splitDate(correctDate);
+        const value = checkAvailability(hour, correctDate);
+
+        newInfo = { ...availabilityInfo };
+
+        if (value === "none") {
+            if (newInfo[dayData[0]] === undefined) {
+                newInfo = { ...newInfo, [dayData[0]]: {} };
+            }
+            if (newInfo[dayData[0]][dayData[1]] === undefined) {
+                newInfo[dayData[0]] = { ...newInfo[dayData[0]], [dayData[1]]: {} };
+            }
+            if (newInfo[dayData[0]][dayData[1]][dayData[2]] === undefined) {
+                newInfo[dayData[0]][dayData[1]] = {
+                    ...newInfo[dayData[0]][dayData[1]],
+                    [dayData[2]]: {},
+                };
+            }
+            newInfo[dayData[0]][dayData[1]][dayData[2]][hour] = "green";
+        }
+        if (value === "green") {
+            newInfo[dayData[0]][dayData[1]][dayData[2]][hour] = "red";
+        }
+        if (value === "red") {
+            newInfo[dayData[0]][dayData[1]][dayData[2]][hour] = "none";
+        }
+        setAvailabilityInfo(newInfo);
+    };
+
+    const checkAvailability = (hour, date) => {
+        const dayInfo = splitDate(date);
+        const year = dayInfo[0];
+        const month = dayInfo[1];
+        const day = dayInfo[2];
+        if (
+            availabilityInfo[year] &&
+            availabilityInfo[year][month] &&
+            availabilityInfo[year][month][day] &&
+            availabilityInfo[year][month][day][hour]
+        ) {
+            return availabilityInfo[year][month][day][hour];
+        }
+        return "none";
+    };
+
+    useEffect(() => {
+        scrollViewRef.current.scrollTo({ y:635, animated: true });
+    });
+    return (
+        <View className="h-full">
+            <View className="h-20 flex-row justify-center items-center">
+                <Text className="font-semibold text-lg">Choose Date:</Text>
+                <DateTimePicker
```

Comment:
bit of a weird one but this is formatted as "6 Jun 2024" 99% of the time then goes to "30/06/2024" for some dates, very randomly